File: src/domain/userId.js
var valueType = require('../valueType');

var UserEmailCannotBeEmpty = exports.UserEmailCannotBeEmpty = function UserEmailCannotBeEmpty(){};

var UserId = exports.UserId = valueType.extends(function UserId(email){
    if(!email){
        throw new UserEmailCannotBeEmpty();
    }

    this.email = email;
}, function toString(){
    return 'UserId:' + this.email;
});

File: Mixter.Domain/Identity/UserId.cs
namespace Mixter.Domain.Identity
{
    public struct UserId
    {
        public string Email { get; private set; }

        public UserId(string email)
            : this()
        {
            if (string.IsNullOrWhiteSpace(email))
            {
                throw new UserEmailCannotBeEmpty();
            }

            Email = email;
        }

        public override string ToString()
        {
            return Email;
        }
    }
}
